app/models/setting.rb

```diff
@@ -285,11 +289,13 @@ def remove_whitespaces
     self[:value] = value.each { |a| a.strip! if a.respond_to? :strip! }
   end
 
-  def encrypt_url_if_password_present
-    uri = URI.parse(value)
-    return unless uri.userinfo&.include?(':')
-
-    self[:value] = encrypt_field(value)
+  # Check if URL contains user:password credentials
+  def update_encrypted_flag_from_url(value)
+    if URI.parse(value).userinfo&.include?(':')
+      setting_definition&.encrypted = true
+    else
+      setting_definition&.encrypted = false
+    end
```

Comment:
Using the safe navigator in an assignment is odd. That would result in `nil = true`. If it's a concern that the definition is `nil` then I'd expect a return before that.

I'd also be tempted to rewrite it to a direct assignment. Only ugly thing is that it can return `nil`, but if we take the above suggestion to always encrypt if any userinfo is present it becomes cleaner:

```suggestion
    setting_definition.encrypted = !URI.parse(value).userinfo.nil?
```

Reply:
Also, thinking more about it I wonder if this isn't an inversion of control. You shouldn't modify the definition. Instead I think the definition should have some method like:
```ruby
def encrypted?
  setting_definition.encrypted? || (setting_definition.settings_type == "url" && !URI.parse(value).userinfo.nil?)
end
```

Or something along those lines. Right now it straight up delegates the method to the definition, but the definition doesn't have the value.

You can consider something in the definition that's like `encrypted_value?(value)` but I'm not sure that's actually better.